tests/test_converters.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+
+from starlette.convertors import CONVERTOR_TYPES, Convertor, add, reset
+from starlette.responses import Response
+from starlette.routing import Router
+from starlette.testclient import TestClient
+
+
+class HexConvertor(Convertor):
+    """Simple hex <-> bytes convertor for testing.
+    """
+
+    regex = r"[0-9A-Fa-f]+"
+
+    def convert(self, value: str) -> bytes:
+        return bytes.fromhex(value)
+
+    def to_string(self, value: bytes) -> str:
+        return value.hex()
+
+
+@pytest.fixture(autouse=True)
+def reset_convertors(request):
+    request.addfinalizer(reset)
```

Comment:
What is `request` and `addfinalizer` here?

Reply:
`request` is a magic Pytest argument to access the “test context”. `request.addfinalizer` is a common pattern to [register clean up code](https://docs.pytest.org/en/latest/fixture.html#fixture-finalization-executing-teardown-code); this fixture essentially calls `reset` after every test to ensure the environment is clean.

Reply:
My two cents: can this also be achieved with a `yield` fixture?

```python
@pytest.fixture(autouse=true)
def reset_converters_after_test():
    yield
    reset()
```

Reply:
I believe so, yes.

Reply:
Good call @florimondmanca  - I'd prefer that, since "request" is *really* ambiguous.